src/Behat/Testwork/ServiceContainer/ExtensionManager.php

```diff
@@ -228,4 +234,29 @@ private function validateExtensionInstance($extension, $locator)
             ), $locator);
         }
     }
+
+    private function discoverExtensionWithPuli($locator)
```

Comment:
You mean abstracting external tools or abstracting the loaders (one for the file loader, one for class locator, one for puli, ... and so on) ?